### PR TITLE
Add HTTP gateway support for GCP

### PR DESCRIPTION
**Pull request by amarthadan**
[Issue AN-308](https://api3dao.atlassian.net/browse/AN-308)

### CODE REVIEW

File: packages/airnode-deployer/src/handlers/gcp/index.ts
Comment (andreogle) [inline packages/airnode-deployer/src/handlers/gcp/index.ts:76]: Why is this?

**Review by andreogle — APPROVED** [review]

File: packages/airnode-deployer/src/handlers/gcp/index.ts
Comment (amarthadan) [reply packages/airnode-deployer/src/handlers/gcp/index.ts:76]: That was a change introduced by @Siegrift to an AWS HTTP gateway, I just wanted to keep it consistent. I think he'll be able to answer.

Comment (Siegrift) [reply packages/airnode-deployer/src/handlers/gcp/index.ts:76]: It's as the comment say. If you did a request in http gw you would have a response:
```
{"value": {"rawValue": ..., "encodedValue": ..., "values": [...]}}
```

that is unnecessary an you isntead want the reponse flattened, like described in docs https://docs.api3.org/airnode/v0.4/grp-providers/guides/build-an-airnode/http-gateway.html#using-curl

File: packages/airnode-deployer/src/handlers/gcp/index.ts
Comment (Siegrift) [inline packages/airnode-deployer/src/handlers/gcp/index.ts:58]: This is here because we can't whitelist by this header in the gateway, right? (in aws this is done by aws)

Comment (Siegrift) [reply packages/airnode-deployer/src/handlers/gcp/index.ts:58]: It would deserve a comment.

File: packages/airnode-deployer/src/handlers/gcp/index.ts
Comment (Siegrift) [reply packages/airnode-deployer/src/handlers/gcp/index.ts:76]: It could be refactored so that this wrapping is removed and the flattened value is only returned.

File: packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf
Comment (Siegrift) [inline packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf:50]: What is this for?

File: packages/airnode-deployer/terraform/airnode/gcp/templates/apigateway.yaml.tpl
Comment (Siegrift) [inline packages/airnode-deployer/terraform/airnode/gcp/templates/apigateway.yaml.tpl:1]: I guess this is for the open API file referenced in terraform, right?

**Review by Siegrift — COMMENTED** [review]
It would be nice (at least for me) to know how this works and why are some things needed (e.g. the openapi spec). It might be worth adding a bit of documentation to the teraform stuff

File: packages/airnode-deployer/src/handlers/gcp/index.ts
Comment (amarthadan) [reply packages/airnode-deployer/src/handlers/gcp/index.ts:58]: Yes, that's why we have to do it manually here. I'll add a comment.

File: packages/airnode-deployer/terraform/airnode/aws/main.tf
Comment (amarthadan) [reply packages/airnode-deployer/terraform/airnode/aws/main.tf:79]: I set `testApi` handler timeout (both AWS and GCP) to 15 because the only thing it does is call the `callAPI` handler which has a timeout of 10. So, anything higher is unnecessary.

File: packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf
Comment (amarthadan) [reply packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf:47]: This is used to uniquely identify an API config. During the deployment of a new version, first, a new config is created and then the old one is destroyed, so, for a few moments, there are two of them, that's why we need to distinguish them.
The ID can have a max length of 64 characters. Taking into consideration our naming convention, I don't really want to make the identifying part longer as it may become a problem. But anyway, the first 8 characters from a UUID should be unique enough to avoid collisions between just two resources.

File: packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf
Comment (amarthadan) [reply packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf:50]: This is the gateway definition. The OpenAPI definition describes what endpoints of what type should be available on the gateway. Pretty much the same as for AWS.

File: packages/airnode-deployer/terraform/airnode/gcp/templates/apigateway.yaml.tpl
Comment (amarthadan) [reply packages/airnode-deployer/terraform/airnode/gcp/templates/apigateway.yaml.tpl:1]: Yes.

**Comment by amarthadan** [comment]
> It would be nice (at least for me) to know how this works and why are some things needed (e.g. the openapi spec). It might be worth adding a bit of documentation to the teraform stuff

I created an [issue](https://api3dao.atlassian.net/browse/AN-508) for writing such documentation.

File: packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf
Comment (Siegrift) [reply packages/airnode-deployer/terraform/airnode/gcp/modules/apigateway/main.tf:50]: I forgot (or didn't know) we have that for AWS as well.

**Review by Siegrift — APPROVED** [review]
LGTM

**Comment by amarthadan** [comment]
Not merging this yet since the issue is for v0.5 (I can create a v0.5 branch and put it there but it's the only v0.5 issue so far so just holding off the merge is good enough for now). But moving the issue to testing in Jira.

**Comment by amarthadan** [comment]
We've decided to include this in v0.4 after all so I'm merging it.

**Pull request merged**
